File: backend/src/models/exercise.model.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db.config');

const Exercise = sequelize.define('Exercise', {
    name: { type: DataTypes.STRING, allowNull: false },
    description: { type: DataTypes.TEXT },
    muscleGroup: { 
        type: DataTypes.ENUM('chest', 'back', 'legs', 'shoulders', 'arms', 'core', 'full_body', 'cardio'),
        allowNull: false 
    },
    equipment: { 
        type: DataTypes.ENUM('none', 'dumbbell', 'barbell', 'kettlebell', 'machine', 'cable', 'bands'),
        allowNull: false 
    },
    videoUrl: { type: DataTypes.STRING }, // YouTube Link
}, { timestamps: true });

module.exports = Exercise;

File: lib/features/workout/data/models/exercise_model.dart
class ExerciseModel {
  final int    id;
  final String name;
  final String description;
  final String muscleGroup;
  final String equipment;
  final String? videoUrl;

  const ExerciseModel({
    required this.id,
    required this.name,
    required this.description,
    required this.muscleGroup,
    required this.equipment,
    this.videoUrl,
  });

  factory ExerciseModel.fromJson(Map<String, dynamic> j) => ExerciseModel(
    id:          j['id']          as int?    ?? 0,
    name:        j['name']        as String? ?? '',
    description: j['description'] as String? ?? '',
    muscleGroup: j['muscleGroup'] as String? ?? '',
    equipment:   j['equipment']   as String? ?? '',
    videoUrl:    j['videoUrl']    as String?,
  );
}